llvm/lib/SYCL/PrepareSYCLOpt.cpp

```diff
@@ -220,13 +222,31 @@ struct PrepareSYCLOpt : public ModulePass {
     }
   }
 
+  struct CheckUnsupportedBuiltinsVisitor
+      : public llvm::InstVisitor<CheckUnsupportedBuiltinsVisitor> {
+    void visitCallInst(CallInst &I) {
+      auto *F = I.getCalledFunction();
+      assert(!(llvm::demangle(std::string(F->getName()))
+                   .rfind("__spir_ocl_get", 0) == 0) &&
+             "Unsupported SPIR builtin in HLS flow");
+    }
+  };
+
+  void checkSPIRBuiltins(Module &M) {
+    CheckUnsupportedBuiltinsVisitor CUBV{};
+    for (auto &F : M.functions()) {
+      CUBV.visit(F);
+    }
+  }
+
   bool runOnModule(Module &M) override {
     // When using the HLS flow instead of SPIR default
     bool SyclHLSFlow = Triple(M.getTargetTriple()).isXilinxHLS();
     unwrapFPGAProperties(M);
     turnNonKernelsIntoPrivate(M);
     if (SyclHLSFlow) {
       setHLSCallingConvention(M);
+      checkSPIRBuiltins(M);
```

Comment:
The name is not precise enough, without comment. Check for what?
`validateAllowedSPIRBuiltins(M);`?
Are there some SPIR intrinsics which are allowed? Or do you assume none?
It changes the naming.

Reply:
As of now only a few spir builtins are checked. 
We need to discuss with @Ralender to see if we can forbid all spir builtins or if some of them (such as mathematical primitives) are still required for linking with HLS libraries.